testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,526 @@
+                        if (ob_0 instanceof ThreadLocal) {
+                            threadLocal = true;
+                        }
+                        try {
+                            // directly invoke reset when dealing with a Mockito mock object
+                            if (Mockito.mockingDetails(ob_0).isMock()) {
+                                Method m = Mockito.class.getDeclaredMethod("reset", Object[].class);
+                                m.invoke(null, new Object[]{new Object[]{ob_0}});
+                            } else {
+                                UnmarshalChain.reset();
+                                UnmarshalChain.initializeChain(fieldList[i].getDeclaringClass().getName(), fieldList[i].getName());
+                                ob_0 = xstream.fromXML(state0);
+                            }
+                        } catch (NoSuchMethodError nsme) {
+                            UnmarshalChain.reset();
+                            UnmarshalChain.initializeChain(fieldList[i].getDeclaringClass().getName(), fieldList[i].getName());
+                            ob_0 = xstream.fromXML(state0);
+                        }
+                        try {
+                            if (AccessibleObject.class.isAssignableFrom(ob_0.getClass())) {
+                                ((AccessibleObject) ob_0).setAccessible(true);
+                            }
+                        } catch (NullPointerException npex) {
+                            npex.printStackTrace();
+                        }
+                        if (threadLocal) {
+                            Object tmp = fieldList[i].get(null);
+                            ((ThreadLocal)tmp).set(ob_0);
+                            ob_0 = tmp;
+                        }
+
+                        FieldUtils.writeField(fieldList[i], (Object)null, ob_0, true);
+
+                        String output = fieldName + " set\n";
+                        Files.write(Paths.get(reflectionFile), output.getBytes(),
+                                StandardOpenOption.APPEND);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        String outputNormalError = fieldName + " reflectionError: " + e + "\n";
+                        Files.write(Paths.get(reflectionFile), outputNormalError.getBytes(),
+                                StandardOpenOption.APPEND);
+                    }
+                }
+            } catch (Exception e) { // *** check the kind of exception
+                e.printStackTrace();
+            }
+        } catch (Exception e) {
+            String output = fieldName + " deserializeError: " + e + "\n";
+            Files.write(Paths.get(reflectionFile), output.getBytes(),
+                    StandardOpenOption.APPEND);
+        }
+    }
+
+    @Override
+    public void capture() {
+        try {
+            String state = Configuration.config().getProperty("statecapture.state", "");
```

Comment:
Can move this assignment below `if`, since it's pointless to do this if we end up returning.